spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Recurrent.scala

```diff
@@ -81,12 +81,13 @@ class Recurrent[T : ClassTag]()
 
   /**
    * Clone N models; N depends on the time dimension of the input
-   * @param times
-   * @param batchSize
-   * @param hiddenSize
+   * @param sizes, the first element is batchSize, the second is times, the third is hiddensize
+    *             the left is size of images
    */
-  private def extend(times: Int, batchSize: Int, hiddenSize: Int,
-    rows: Int = 1, columns: Int = 1): Unit = {
+  private def extend(sizes: Array[Int]): Unit = {
+    val times = sizes(1)
+    val batchSize = sizes(0)
+    val imageSize = sizes.drop(3)
```

Comment:
Actually array used in the change is quite small, at most 24 bytes, I doubted if it will cause perf issue with minor gc as we can see lots of small arrays are created with tensor operations.